.github/workflows/test.yaml

```diff
@@ -46,6 +46,46 @@ jobs:
           name: execution-report
           path: ./test-execution.xml
 
+      # - name: Test Report
+      #   uses: dorny/test-reporter@v1
+      #   if: always()
+      #   with:
+      #     name: Test Results
+      #     path: test-execution.xml
+      #     reporter: java-junit
+      #     token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Save badge env variables
+        run: |
+          echo "TEST_COUNT=$(grep -o '<testcase ' test-execution.xml | wc -l)" >> $GITHUB_ENV
+          echo "TEST_PASSED=$(($(grep -o '<testcase ' test-execution.xml | wc -l) - $(grep -o '<failure ' test-execution.xml | wc -l)))" >> $GITHUB_ENV
+          echo "COVERAGE=$(grep -o 'line-rate="[^"]*"' coverage.xml | head -1 | cut -d'"' -f2 | awk '{printf "%.2f", $1 * 100}')%" >> $GITHUB_ENV
+          echo "COVERAGE_COLOR=$(if [ $(echo $COVERAGE | cut -d'.' -f1) -ge 80 ]; then echo 'green'; elif [ $(echo $COVERAGE | cut -d'.' -f1) -ge 50 ]; then echo 'orange'; else echo 'red'; fi)" >> $GITHUB_ENV
+
+      - name: Test Count Badge
+        uses: RubbaBoy/BYOB@v1.3.0
+        with:
+          name: test-badge
+          label: Tests
+          status: "${{ env.TEST_PASSED }} / ${{ env.TEST_COUNT }} passed"
+          color: 'green'
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Coverage Badge
+        uses: RubbaBoy/BYOB@v1.3.0
+        with:
+          name: coverage-badge
```

Comment:
For this part, because the `GITHUB_TOKEN` is a sensitive item, I would try to steer clear of passing it to any Github Action that isn't being actively maintained, and doesn't have a lot of users.

https://shields.io/badges/git-hub-actions-workflow-status

We could use something like this ^ perhaps, thoughts?

Reply:
Good point. Let me see what I can do as an alternative.

Reply:
I've replaced with dynamic badges from shields.io and a manual update to the repo file. No secrets required.